hibernate-core/src/main/java/org/hibernate/ForcedFlushMode.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate;
+
+/**
+ * Enumerates the possible flush modes for execution of a
+ * {@link org.hibernate.query.Query}. A "forced" flush mode
+ * overrides the {@linkplain Session#getHibernateFlushMode()
+ * flush mode of the session}.
+ *
+ * @author Gavin King
+ *
+ * @since 6.2
+ */
+public enum ForcedFlushMode {
+	/**
+	 * Flush before executing the query.
+	 */
+	FORCE_FLUSH,
+	/**
+	 * Do not flush before executing the query.
+	 */
+	FORCE_NO_FLUSH,
+	/**
+	 * Let the owning {@link Session session} decide whether
+	 * to flush, depending on its {@link FlushMode}.
+	 */
+	NO_FORCING
+}
```

Comment:
Everyone, please take a look at this new thing!

Reply:
Given that this is solely about queries, I'd be more inclined to name this `QueryFlushMode` or move it into `o.h.query`.  For `QueryFlushMode`, I'd go with:

```
enum QueryFlushMode {
    FLUSH,
    SKIP,
    /**
     * Defer to {@link Session#getHibernateFlushMode}
     */
    AUTO
}
```

I get why you named it as you did.  I just find it confusing

Reply:
1. `QueryFlushMode` is fine by me, it was also my first thought.
2. You're right, I agree it belongs in `o.h.query`.
3. `FLUSH` is fine.
4. `SKIP` I don't find _extremely_ clear though I could live with it.
5. I don't like `AUTO` here because it doesn't have the same semantics as `FlushMode.AUTO`.